packages/remix-server-runtime/server.ts

```diff
@@ -360,12 +360,12 @@ async function handleDataRequest(
 
     // Mark all successful responses with a header so we can identify in-flight
     // network errors that are missing this header
-    response.headers.set("X-Remix-Response", "yes");
```

Comment:
If this response is a direct `undici` `fetch` `Response`, then the headers are immutable and we can't change them.  I didn't want to clone the response in 100% of scenarios since this isn't an issue with `return json()` and alternatives, so I am only cloning if this operation throws the `immutable` error